src/runtime/providers/string.ts

```diff
@@ -125,6 +124,90 @@ class Substring extends Constraint {
   }
 }
 
+class Find extends Constraint {
+  static AttributeMapping = {
+    "text": 0,
+    "subtext": 1,
+    "case-sensitive": 2,
+    "from": 3,
+  }
+  static ReturnMapping = {
+    "string-position": 0,
+    "result-index": 0,
+  }
+
+  returnType: "both" | "position";
+
+  constructor(id: string, args: any[], returns: any[]) {
+    super(id, args, returns);
+    if(this.returns[1] !== undefined && this.returns[0] !== undefined) {
+      this.returnType = "both"
+    } else if(this.returns[0] !== undefined) {
+      this.returnType = "position";
+    }
```

Comment:
I don't think it's very important, but it seems reasonable to want just the result-index if you're just counting occurrences or something.